docker/entrypoint.sh

```
@@ -11,17 +11,45 @@ if [ -n "$DB_HOST" ]; then
+# The key is persisted in the storage volume (shared with the scheduler container)
+# so restarts and container recreation keep using the same key instead of
+# invalidating existing sessions / encrypted data.
+if [ -z "$APP_KEY" ]; then
+  APP_KEY_FILE="/var/www/html/storage/app/.app_key"
+  if [ -f "$APP_KEY_FILE" ]; then
+    APP_KEY=$(cat "$APP_KEY_FILE")
+  else
+    echo "No APP_KEY set, generating one..."
+    APP_KEY=$(php artisan key:generate --show --no-interaction)
+    mkdir -p "$(dirname "$APP_KEY_FILE")"
+    echo -n "$APP_KEY" > "$APP_KEY_FILE"
```

Shellcheck (0.11.0)

[warning] In POSIX sh, echo flags are undefined.

(SC3037)